### PR TITLE
x86_64: get enough things working to enable full `start.zig` logic

### PR DESCRIPTION
**Pull request by jacobly0**
This reaches the milestone of being able to run [Hello World](https://ziglang.org/documentation/master/#Hello-World) without modification.
```
$ cat hello.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    try stdout.print("Hello, {s}!\n", .{"world"});
}
$ zig run hello.zig -fno-LLVM
Hello, world!

### CODE REVIEW

**Comment by kubkon** [comment]
@Vexu could you take a look at the incremental compilation crashes and help debug those please? I am out of my depth here.

**Comment by jacobly0** [comment]
I have done enough debugging to understand what sequence of events is causing incremental compilation to crash, but I don't understand the system well enough to know which step is actually going wrong.  All of my attempted fixes have only led to more crashes, so it's hard to tell if any of them were actual progress.

**Comment by Vexu** [comment]
Those errors look similar to ones I've had with bad caches, not sure how that could be the cause though.

File: test/cases/x86_64-linux/hello_world_with_updates.0.zig [hunk 1]
```diff
@@ -2,4 +2,7 @@
 // output_mode=Exe
 // target=x86_64-linux
 //
-// :110:9: error: root struct of file 'tmp' has no member named 'main'
+// :604:45: error: root struct of file 'tmp' has no member named 'main'
+// :553:12: note: called from here
+// :503:36: note: called from here
+// :466:17: note: called from here
```
Comment (Vexu) [inline test/cases/x86_64-linux/hello_world_with_updates.0.zig:8]:
```suggestion
// :?:?: error: root struct of file 'tmp' has no member named 'main'
// :?:?: note: called from here
// :?:?: note: called from here
// :?:?: note: called from here
```
And same with the other.

**Pull request merged**